resp.go

```diff
@@ -201,46 +204,48 @@ func readI(i *bufio.Reader) (v int64, err error) {
 	return v * s, nil
 }
 
-func readB(i *bufio.Reader) (string, error) {
+func readB(i *bufio.Reader) (*byte, int64, error) {
 	length, err := readI(i)
 	if err != nil {
-		return "", err
+		return unsafe.StringData(""), 0, err
 	}
 	if length == -1 {
-		return "", errOldNull
+		return unsafe.StringData(""), 0, errOldNull
 	}
 	bs := make([]byte, length)
 	if _, err = io.ReadFull(i, bs); err != nil {
-		return "", err
+		return unsafe.StringData(""), 0, err
 	}
 	if _, err = i.Discard(2); err != nil {
-		return "", err
+		return unsafe.StringData(""), 0, err
 	}
-	return BinaryString(bs), nil
+	return unsafe.SliceData(bs), int64(len(bs)), nil
 }
 
-func readE(i *bufio.Reader) ([]RedisMessage, error) {
-	v := make([]RedisMessage, 0)
+func readE(i *bufio.Reader) (*RedisMessage, int64, error) {
+	v := unsafe.SliceData(make([]RedisMessage, 0))
+	length := 0
 	for {
 		n, err := readNextMessage(i)
 		if err != nil {
-			return nil, err
+			return nil, 0, err
 		}
 		if n.typ == '.' {
-			return v, err
+			return v, int64(length), err
 		}
-		v = append(v, n)
+		v = unsafe.SliceData(append(unsafe.Slice(v, length), n))
+		length++
 	}
 }
 
-func readA(i *bufio.Reader, length int64) (v []RedisMessage, err error) {
-	v = make([]RedisMessage, length)
+func readA(i *bufio.Reader, length int64) (v *RedisMessage, l int64, err error) {
+	v = unsafe.SliceData(make([]RedisMessage, length))
```

Comment:
Can we keep v := make([]RedisMessage, length) and only extract the SliceData when returning?